### PR TITLE
use size from datastore and check to be not more than provided size

### PR DESCRIPTION
**Pull request by giggsoff**
We must have size of blob to download parts of it correctly. But seems current logic has no checks for size provided from configuration. In this case we can hit the problems when we have miss-configured size, so lets get size from datastore in all cases and check for mismatch.

### CODE REVIEW

File: libs/zedUpload/awsutil/s3sync.go [hunk 1]
```diff
@@ -245,17 +245,19 @@ func getNeededParts(cWriterOptions *writerOptions, bname, bkey string, doneParts
...
-			return doneParts, err
-		}
+	err, bsize := s.GetObjectSize(bname, bkey)
+	if err != nil {
+		return doneParts, err
+	}
+
+	if objMaxSize != 0 && bsize > objMaxSize {
+		return types.DownloadedParts{PartSize: S3PartSize},
+			fmt.Errorf("configured image size (%d) is less than blob size (%d)", objMaxSize, bsize)
```
Comment (zededa-yuri) [inline libs/zedUpload/awsutil/s3sync.go:260]: is "blob size" a correct wording here? Would it make sense rather say something like "Configured volume size is smaller the the size of image associated with it"?

**Review by zededa-yuri — APPROVED** [review]
LGTM with one NIT

**Review by zededa-yuri — APPROVED** [review]
LGTM

**Review by rvs — APPROVED** [review]
LGTM

**Comment by eriknordmark** [comment]
I recall that in tjhe past we've had images (perhaps it was container/OCI images) where the configured size in the controller was set to zero. How will those be handled with this fix in place?

I did a quick check in a test enterprise using
zcli image show | grep '0.0 BB' | grep Ready
and there are lots of such test containers.

File: libs/zedUpload/awsutil/s3sync.go [hunk 2]
```diff
@@ -245,17 +245,19 @@ func getNeededParts(cWriterOptions *writerOptions, bname, bkey string, doneParts
...
-		err, bsize = s.GetObjectSize(bname, bkey)
-		if err != nil {
-			return doneParts, err
-		}
+	err, bsize := s.GetObjectSize(bname, bkey)
+	if err != nil {
+		return doneParts, err
+	}
+
+	if objMaxSize != 0 && bsize > objMaxSize {
```
Comment (eriknordmark) [inline libs/zedUpload/awsutil/s3sync.go:258]: Ah - you handle zero as a wildcard here. OK

**Review by eriknordmark — APPROVED** [review]
LGTM

**Pull request merged**